### PR TITLE
add beginner friendly API documentation #6086

### LINKED ISSUE #3319: Is that possible to add an another example for downloading file with curl?
http://guides.dataverse.org/en/latest/api/sword.html gives a fairly comprehensive usage of the curl command. I'm wondering if there could be another example for accessing a file given filename, doi using curl?

### LINKED ISSUE #3624: correct docs for `api/dataverses/$id/metadatablocks` API endpoint
Current docs have "The query body is a JSON array with a list of metadatablocks identifiers (either id or name)."; but `curl -d $json_list` fails with:
{"status":"ERROR","message":"Invalid token=EOF at (line no=1, column no=0, offset=-1). Expected tokens are: [CURLYOPEN, SQUAREOPEN]"}

`--upload-file $json_list_in_file` works as expected; but it's a field of the POST body (not the entire thing).

### LINKED ISSUE #6086: Beginner friendly API documentation
As I mentioned in the "Upcoming talk on the Dataverse API" thread on the mailing list, I'm adding some more introductory material to the guides to explain what Dataverse APIs are and how they can be used by support teams and others are not deeply technical and programmer-y: https://groups.google.com/d/msg/dataverse-community/V5WkMGDS4VI/maxXTdmzDwAJ 

I don't plan to create a ton of diagrams but here's an example of what I've come up with so far:

Screen Shot 2019-08-12 at 10 22 03 AM

I'm constantly deploying some proposed changes the the one above to this server if anyone would like a sneak peek: https://dev2.dataverse.org/guides/

I may or may not try to address other smallish issues flagged as "API Guide" while I'm in there: https://github.com/IQSS/dataverse/labels/Feature%3A%20API%20Guide

Here's a screenshot of those many issues:

Issues_·_IQSS_dataverse_-_2019-08-12_10 24 01

### PR DESCRIPTION
**Pull request by pdurbin**
- Closes #6086 (Beginner friendly API documentation)
- Also closes #3319 (Is that possible to add an another example for downloading file with curl?)
- Also closes #3624 correct docs for `api/dataverses/$id/metadatablocks` API endpoint  

---

To give feedback on this pull request, you are welcome to look at the following build of the guides:

http://guides.dataverse.org/en/6086-api-guide/api/intro.html

Here's a screenshot of the re-written "intro" page as of 39cd360

Screen Shot 2019-08-20 at 3 22 57 PM

Update for 2019-08-20:

- After standup we discussed the scope of this issue. I showed a longish list of related issues but we decided only to add "closes" for the issues above. For the record, here are the open issues I was considering tackling to some degree or another:
    -  SWORD API documentation cleanup #3588 
    -  curl issue with native API #4510 
    -  native API for creating dataverse reports parse error for valid json #3533 
    - doi, "start" parameter using the Search API #4377 
- We also discussed two use cases from @sbarbosadataverse that came out of my API talk. I chatted with her afterwards and she plans to make issues for them:
    - I want to download many files at once.
    - I want to upload many files each with their own description at once.
- Also on the screen from my todo list was feedback from the community:
    - document DatasetThumbnail API
    - add section on undocumented APIs? See Document - Make use of MyData API “official” #5042
    - one thing, that i asked myself: is it mandatory to add the typeClass and multiple attributes to the Dataset JSON?
    - and which attributes are required in the JSON file, for Dataverse, Dataset and Datafile.
    - and: which fields can only have one value, and which can have multiple (arrays)?
- In short, there's agreement that the pull request improves the API Guide but we need to "trim the tail" and save the work above for another day. Pull requests welcome! I'd also like to highlight that @donsizemore mentioned that a graduate student might be able to help with #6083 which is about moving to full curl examples, especially for `POST` commands where you need to send JSON or whatever. I hope people like the way I'm documenting things now in the files I touched, with Bash `export` commands for environment variables used in curl commands. Now color coded! Now with lines that wrap! 😄 
- Oh, I also asked for feedback from the community at https://groups.google.com/d/msg/dataverse-community/V5WkMGDS4VI/PLzPUVvoFwAJ

### CODE REVIEW

File: doc/sphinx-guides/source/user/account.rst [hunk 1]
```diff
@@ -164,5 +164,33 @@ Notifications will only be emailed one time even if you haven't read the notific
...
+
+How Your API Token Is Like a Password
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In many cases, such as when depositing data, an API Token is required to interact with Dataverse APIs. The word "token" indicates a series of letters and numbers such as ``c6527048-5bdc-48b0-a1d5-ed1b62c8113b``. Anyone who has your API Token can add and delete data as you so you should treat it with the same care as a password.
+
+How to Create Your API Token
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To create your API token, click on your name in the upper right and then click "API Token". In this tab, click "Create Token".
```
Comment (mheppler) [inline doc/sphinx-guides/source/user/account.rst]: We should avoid saying things like "upper right" or other places on the page you can find features, in case things move around in designs, as well as in responsive layouts in smaller browser viewports. I think something like "..._click on your name in the navbar, then select "API Token" from the dropdown menu._"

File: doc/sphinx-guides/source/api/getting-started.rst [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+~~~~~~~~~~~~~~~~~
+
+The :doc:`dataaccess` section explains how to download files.
+
+In order to download files, you must know their database IDs which you can get from the ``dataverse_json`` metadata at the dataset level. See :ref:`export-dataset-metadata-api`.
+
+Downloading Metadata
+~~~~~~~~~~~~~~~~~~~~
+
+Dataset metadata is availabe in a variety of formats listed at :ref:`metadata-export-formats`.
```
Comment (mheppler) [inline doc/sphinx-guides/source/api/getting-started.rst]: Typo: "_available_"

File: doc/sphinx-guides/source/api/getting-started.rst [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+Getting an API Token
+--------------------
+
+Many Dataverse APIs require an API token.
+
+Once you have identified a server to test with, create an account, click on your name, and get your API token. For more details, see the :doc:`auth` section.
+
+.. _curl-examples-and-environment-variables:
+
+curl Examples and Enviroment Variables
```
Comment (mheppler) [inline doc/sphinx-guides/source/api/getting-started.rst]: Typo: "_Environment_"

File: doc/sphinx-guides/source/api/faq.rst [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+It's possible to get a complete list of API functionality in Swagger/OpenAPI format if you deploy Dataverse to Payara 5+. For details, see https://github.com/IQSS/dataverse/issues/5794
+
+Is There a Changelog of API Functionality That Has Been Added Over Time?
+------------------------------------------------------------------------
+
+No, but there probably should be. If you have suggestions for how it should look, please create an issue at https://github.com/IQSS/dataverse/issues
+
+.. _no-api:
+
+What Funtionality is GUI Only and Not Available Via API
```
Comment (mheppler) [inline doc/sphinx-guides/source/api/faq.rst]: Typo: "_Functionality_"

File: doc/sphinx-guides/source/api/auth.rst [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+An API token is similar to a password and allows you to authenticate to Dataverse APIs to perform actions as you. Many Dataverse APIs require the use of an API token.
+
+.. contents:: |toctitle|
+    :local:
+
+How to Get an API Token
+-----------------------
+
+Your API token is unique to the server you are using. You cannot use your API token from one server on another server.
```
Comment (mheppler) [inline doc/sphinx-guides/source/api/auth.rst]: More better: "_You cannot take your API token from one server and use it on another server._"

**Comment by mheppler** [comment]
Fixed typo's and other small details as promised. Ready for QA.

**Comment by pdurbin** [comment]
@mheppler thanks for fixing those typos!

File: doc/sphinx-guides/source/admin/troubleshooting.rst [hunk 6]
```diff
@@ -3,11 +3,35 @@
...
 
 .. contents:: Contents:
 	:local:
 
+Using Dataverse APIs to Troubleshoot and Fix Problems
+-----------------------------------------------------
+
+See the :doc:`/api/intro` section of the API Guide for a high level overview of Dataverse APIs. Below are listed problems that support teams might encounter that can be handled via API (sometimes only via API).
+
+A Dataset Is Locked And Cannot Be Edited or Published
```
Comment (donsizemore) [inline doc/sphinx-guides/source/admin/troubleshooting.rst]: It might be helpful to point out that one should be certain Dataverse is finished with a dataset (worst case: stop glassfish) before removing the lock.

File: doc/sphinx-guides/source/admin/troubleshooting.rst [hunk 7]
```diff
@@ -3,11 +3,35 @@
...
+It's normal for the ingest process described in the :doc:`/user/tabulardataingest/ingestprocess` section of the User Guide to take some time but if hours or days have passed and the dataset is still locked, you might want to inspect the locks and consider deleting some or all of them.
+
+See :doc:`dataverses-datasets`.
+
+Someone Created Spam Datasets and I Need to Delete Them
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Depending on how open your installation of Dataverse is to the general public creating datasets, you may sometimes need to deal with spam datasets.
+
+Look for "destroy" in the :doc:`/api/native-api` section of the API Guide.
```
Comment (donsizemore) [inline doc/sphinx-guides/source/admin/troubleshooting.rst]: Point out that unpublished datasets may be deleted  via http://guides.dataverse.org/en/latest/api/native-api.html#delete-unpublished-dataset ?

File: doc/sphinx-guides/source/api/auth.rst [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+An API token is similar to a password and allows you to authenticate to Dataverse APIs to perform actions as you. Many Dataverse APIs require the use of an API token.
+
+.. contents:: |toctitle|
+    :local:
+
+How to Get an API Token
+-----------------------
+
+Your API token is unique to the server you are using. You cannot take your API token from one server and use it on another server.
```
Comment (donsizemore) [inline doc/sphinx-guides/source/api/auth.rst]: s/server/Dataverse instance/g ?

File: doc/sphinx-guides/source/api/auth.rst [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+Instructions for getting a token are described in the :doc:`/user/account` section of the User Guide.
+
+How Your API Token Is Like a Password
+-------------------------------------
+
+Anyone who has your API Token can add and delete data as you so you should treat it with the same care as a password.
```
Comment (donsizemore) [inline doc/sphinx-guides/source/api/auth.rst]: ... add and delete data with your credentials, so you should ?

File: doc/sphinx-guides/source/api/getting-started.rst [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+See :ref:`export-dataset-metadata-api`.
+
+Listing the Contents of a Dataverse
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+See :ref:`show-contents-of-a-dataverse-api`.
+
+Managing Permissions
+--------------------
+
```
Comment (donsizemore) [inline doc/sphinx-guides/source/api/getting-started.rst]: don't see a good target here. TBD?
